source/containers.tex

```diff
@@ -7798,6 +7840,1109 @@
+void trim_capacity(size_type n) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+For the first overload, all reserved blocks are deallocated, and
+\tcode{capacity()} is reduced accordingly.
+For the second overload, \tcode{capacity()} is reduced to no less than \tcode{n}.
+
+\pnum
+\complexity
+Linear in the number of reserved blocks deallocated.
+
+\pnum
+\remarks
+All references, pointers, and iterators referring to elements in \tcode{*this},
+as well as the past-the-end iterator, remain valid.
+\end{itemdescr}
+
+\indexlibrarymember{block_capacity_limits}{hive}%
+\begin{itemdecl}
+constexpr hive_limits block_capacity_limits() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\exposid{current-limits}.
+
+\pnum
+\complexity
+Constant.
+\end{itemdescr}
+
+\indexlibrarymember{block_capacity_default_limits}{hive}%
+\begin{itemdecl}
+static constexpr hive_limits block_capacity_default_limits() noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+A \tcode{hive_limits} struct
+with the \tcode{min} and \tcode{max} members set to
+the implementation's default limits.
+
+\pnum
+\complexity
+Constant.
+\end{itemdescr}
+
+\indexlibrarymember{block_capacity_hard_limits}{hive}%
+\begin{itemdecl}
+static constexpr hive_limits block_capacity_hard_limits() noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+A \tcode{hive_limits} struct
+with the \tcode{min} and \tcode{max} members set to
+the implementation's hard limits.
+
+\pnum
+\complexity
+Constant.
+\end{itemdescr}
+
+\indexlibrarymember{reshape}{hive}%
+\begin{itemdecl}
+void reshape(hive_limits block_limits);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\tcode{T} is \oldconcept{MoveInsertable} into \tcode{hive}.
+
+\pnum
+\effects
+For any active blocks not within the bounds of \tcode{block_limits},
+the elements within those active blocks are reallocated
+to new or existing element blocks which are within the bounds.
+Any element blocks not within the bounds of \tcode{block_limits}
+are deallocated.
+If an exception is thrown during allocation of a new element block,
+\tcode{capacity()} may be reduced,
+reallocation may occur, and
+\exposid{current-limits} may be assigned
+a value other than \tcode{block_limits}.
+Otherwise \tcode{block_limits} is assigned to \exposid{current-limits}.
+If any other exception is thrown the effects are unspecified.
+
+\pnum
+\ensures
+\tcode{size()} is unchanged.
+
+\pnum
+\complexity
+Linear in the number of element blocks in \tcode{*this}.
+If reallocation happens, also linear in the number of elements reallocated.
+
+\pnum
+\remarks
+This operation may change \tcode{capacity()}.
+If reallocation happens, the order of the elements in \tcode{*this} may change.
+Reallocation invalidates all references, pointers, and iterators
+referring to the elements in \tcode{*this},
+as well as the past-the-end iterator.
+\begin{note}
+If no reallocation happens, they remain valid.
```

Comment:
Who is *they*?

Reply:
references, pointers, and iterators

Reply:
Ok nvm, I think it's clear enough in that wording.